tutorial/2_image_clustering.ipynb

```diff
@@ -0,0 +1,524 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x7f323eb134e0>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "print(\"The first image of img_train:\\n\")\n",
+    "plt.imshow(img_train[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To train a PQ-encoder, we pick up the top 1000 images from `img_train`. The clustering will be run on the top 5000 images from `img_test`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "img_train.shape:\n",
+      "(1000, 32, 32, 3)\n",
+      "img_test.shape:\n",
+      "(5000, 32, 32, 3)\n"
+     ]
+    }
+   ],
+   "source": [
+    "img_train = img_train[0:1000]\n",
+    "img_test = img_test[0:5000]\n",
+    "print(\"img_train.shape:\\n{}\".format(img_train.shape))\n",
+    "print(\"img_test.shape:\\n{}\".format(img_test.shape))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Extract a deep feature (VGG16 fc6 activation) from each image using Keras\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, let us extract a 4096-dimensional deep feature from each image. For the feature extactor, we employ an activation from the 6th full connected layer (in Keras implementation, it is called `fc1`) of the ImageNet pre-trained VGG16 model. See the [tutorial of keras](https://keras.io/applications/) for more details. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from keras.applications.vgg16 import VGG16\n",
+    "from keras.applications.vgg16 import preprocess_input\n",
+    "from keras.models import Model\n",
+    "from scipy.misc import imresize\n",
+    "\n",
+    "base_model = VGG16(weights='imagenet')  # Read the ImageNet pre-trained VGG16 model\n",
```

Comment:
It's recommended to employ `include_top=False` option when you only need bottleneck feature.